src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -0,0 +1,1151 @@
+                        string defaultEnumFlagString = string.Join(", ", defaultEnumFlagValues);
+
+                        if (o.ToString() == defaultEnumFlagString)
+                            return;
+                    }
+                    else
+                    {
+                        if (o.ToString() == defaultValue.ToString())
+                            return;
+                    }
+                }
+                else
+                {
+                    if (IsDefaultValue(mapping, o, defaultValue, isNullable))
+                    {
+                        return;
+                    }
+                }
+            }
+
+            XmlQualifiedName xmlQualifiedName = null;
+            if (writeXsiType)
+            {
+                xmlQualifiedName = new XmlQualifiedName(mapping.TypeName, mapping.Namespace);
+            }
+
+            string stringValue = null;
+            bool hasValidStringValue = false;
+            if (mapping is EnumMapping)
+            {
+                stringValue = WriteEnumMethod((EnumMapping)mapping, o);
+                hasValidStringValue = true;
+            }
+            else
+            {
+                hasValidStringValue = WritePrimitiveValue(typeDesc, o, isElement, out stringValue);
+            }
+
+            if (hasValidStringValue)
+            {
+                if (method.HasFlag(WritePrimitiveMethodRequirement.WriteElementString))
+                {
+                    if (method.HasFlag(WritePrimitiveMethodRequirement.Raw))
+                    {
+                        WriteElementString(name, ns, stringValue, xmlQualifiedName);
+                    }
+                    else
+                    {
+                        WriteElementStringRaw(name, ns, stringValue, xmlQualifiedName);
+                    }
+                }
+
+                else if (method.HasFlag(WritePrimitiveMethodRequirement.WriteNullableStringLiteral))
+                {
+                    if (method.HasFlag(WritePrimitiveMethodRequirement.Raw))
+                    {
+                        WriteNullableStringLiteral(name, ns, stringValue);
+                    }
+                    else
+                    {
+                        WriteNullableStringLiteralRaw(name, ns, stringValue);
+                    }
+                }
+                else if (method.HasFlag(WritePrimitiveMethodRequirement.WriteAttribute))
+                {
+                    WriteAttribute(name, ns, stringValue);
+                }
+                else
+                {
+                    // #10593: Add More Tests for Serialization Code
+                    Debug.Assert(false);
+                }
+            }
+            else if (o is byte[])
+            {
+                byte[] a = (byte[])o;
+                if (method.HasFlag(WritePrimitiveMethodRequirement.WriteElementString) && method.HasFlag(WritePrimitiveMethodRequirement.Raw))
```

Comment:
>method.HasFlag(WritePrimitiveMethodRequirement.WriteElementString [](start = 20, length = 65)

`((method & WritePrimitiveMethodRequirement.WriteElementString) != 0)` will be faster. HasFlag does type checking and a null check which will slow things down.